pkg/app/flag.go

```diff
@@ -0,0 +1,46 @@
+package app
+
+import (
+	"os"
+	"path/filepath"
+
+	"github.com/pkg/errors"
+)
+
+const (
+	flagsDir = "/tmp/flags"
+	// LimitedCache represent the flag cache couldn't mount on ssd or hdd
+	LimitedCache = "limited-cache"
+)
+
+// SetFlag is used when the /var/cache cannot be mounted on a SSD or HDD,
+// it will mount the cache disk on a temporary file system in memory.
+func SetFlag(key string) error {
+	// creating a file that will be used as a flag
+	// the flag can 'warn' other deamons that the /var/cache is not on HDD or SDD
+	if err := os.MkdirAll(flagsDir, os.ModePerm); err != nil {
+		return errors.Wrap(err, "failed to create flags directory")
+	}
+
+	f, err := os.Create(filepath.Join(flagsDir, key))
+	if err != nil {
+		return errors.Wrap(err, "failed to create the flag file")
+	}
+
+	return f.Close()
+}
+
+// CheckFlag checks the status of a flag based on a key
+func CheckFlag(key string) bool {
+	_, err := os.Stat(filepath.Join(flagsDir, key))
+	return !os.IsNotExist(err)
+}
+
+// DeleteFlag deletes (unsets) a given flag based on a key
+func DeleteFlag(key string) error {
```

Comment:
this function is vulnerable to "path" injection. I can path a key as "." and then the filepath.Join(flagsDir, key) will return "flagsDir" itself witch will cause the function to delete the entire flagsDir director. A more dangerous attack can be done by setting key to "../" or other calls that can cause the function to delete specific paths on the system knowing how to go there from the flagsDir. 

A check can be as simple as making sure the key doesn't have special "/" character. Or making sure that the final file path to delete is always under flagsDir.